### ISSUE TITLE
Variables for commands

### ISSUE THREAD

**Issue by jzelinskie** [grego/caretaker#3]
Hi there!

I just discovered this project from the [Blades documentation].
The other weekend, I wrote something to scratch my "watch a directory and execute a command" itch -- [reflexd].

I'd love to throw it out entirely and adopt caretaker, but I think I would need caretaker to inject some environment variables with context about the filesystem event occurred. Have you given this any thought in the past?

[Blades documentation]: https://www.getblades.org
[reflexd]: https://github.com/jzelinskie/reflexd

**Comment by grego** [comment]
I'm not sure how it would work with environment variables. Many filesystem events can happen in a short time, overwritting the variables.

A better option IMO would be to provide each command with some variables about the event that it could use.
E.g. if file `foo/bar` changed and the provided command was `cat {}`, it would be turned to `cat foo/bar`.

**Comment by jzelinskie** [comment]
That's exactly my recommendation, sorry for the poor communication. I meant that the implementation of the variables passed to the command could be environment variables, but they could easily not be (my Python project uses the Python 3 format syntax).

Strawman example of environment variable usage:

```toml
[[watch]]
name = "print hello"
path = "src/*"
command = "echo \"hello from $CARETAKER_EVENT_PATH\""
```

**Comment by grego** [comment]
Added the `$EVENT_PATH` environment variable in 58f4a47 and published v0.2.0.
